From: hololeap <hololeap@protonmail.com>
Signed-off-by: hololeap <hololeap@protonmail.com>

Add `executable` USE flag to toggle building the `hnix` executable
(disabled by default)

diff -urN hnix-0.14.0.2/hnix.cabal hnix-0.14.0.2-new1/hnix.cabal
--- hnix-0.14.0.2/hnix.cabal	2021-10-30 18:23:23.462114066 -0600
+++ hnix-0.14.0.2-new1/hnix.cabal	2021-10-30 18:33:56.098568264 -0600
@@ -340,6 +340,10 @@
   manual: True
   default: False
 
+flag executable
+  description: Build the hnix executable
+  default: False
+
 library
   exposed-modules:
     Nix
@@ -499,6 +503,8 @@
   default-language: Haskell2010
 
 executable hnix
+  if !flag(executable)
+    buildable: False
   hs-source-dirs:
     main
   main-is: Main.hs
